### PR TITLE
[release-1.8] :bug: Improve clustermodule existence check

### PR DESCRIPTION
**Pull request by rikatz**
Cherry-pick of #2535  for release 1.8

### CODE REVIEW

**Comment by rikatz** [comment]
/assign @sbueringer 

@chrischdi @sbueringer I will try to make the cherry-pick for 1.7 tomorrow, just waiting this one to pass to make sure I didn't broke anything between releases

File: pkg/util/errors.go [hunk 1]
```diff
@@ -1,26 +0,0 @@
...
-*/
-
-package util
-
-import (
-	"net/http"
-	"strings"
-)
-
-func IsNotFoundError(err error) bool {
```
Comment (chrischdi) [inline pkg/util/errors.go:24]: I think we should keep this around in the old version.

Comment (sbueringer) [reply pkg/util/errors.go:24]: I think we don't have to be as strict as in core CAPI. In core CAPI we have to assume that every exported func can be used by someone importing CAPI as a library.

I wouldn't be as strict in CAPV

**Comment by sbueringer** [comment]
/lgtm

I'm fine either way regarding the public func

**Comment by chrischdi** [comment]
If stefans okay then I am too

/approve

**Pull request merged**